clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredLoginCallbackHandler.java

```diff
@@ -101,6 +103,12 @@
     private Map<String, String> moduleOptions = null;
     private boolean configured = false;
 
+    /** Statically compiled pattern for replacing an escaped double quote. */
+    private static final Pattern BSLASH_DOUBLEQUOTE_REPLACE = Pattern.compile("\"", Pattern.LITERAL);
+
+    /** Statically compiled pattern for replacing 2 backslashes. */
+    private static final Pattern DOUBLE_BSLASH_REPLACE = Pattern.compile("\\", Pattern.LITERAL);
```

Comment:
This one matches a single backslash, not a double backslash. Also, I think you don't need to abbreviate 'backslash'.